cpp/CMakeLists.txt

```diff
@@ -889,8 +889,8 @@ if(NOT MSVC_TOOLCHAIN)
   list(APPEND ARROW_SHARED_LINK_LIBS ${CMAKE_DL_LIBS})
 endif()
 
-set(ARROW_TEST_LINK_TOOLCHAIN arrow::flatbuffers ${ARROW_GTEST_GTEST_MAIN}
-                              ${ARROW_GTEST_GTEST} ${ARROW_GTEST_GMOCK})
+set(ARROW_TEST_LINK_TOOLCHAIN arrow::flatbuffers ${ARROW_GTEST_GMOCK}
```

Comment:
Why can we not use the gtest targets directly here?

Reply:
Because `GTest::gmock` and `GTest::gtest_main` depend on `GTest::gtest`.
`GTest::gtest` is used automatically by specifying only `GTest::gmock` or `GTest::gtest_main`.

Reply:
`${ARROW_GTEST_GMOCK}` <- I was more talking about the use of a variable here instead of the target directly rather which target specifically

Reply:
Ah, I see. Sorry.
We use `arrow::GTest::*` for bundled GoogleTest and `Gtest::*` for system GoogleTest.
So we refer them via `ARROW_GTEST_*` variable.